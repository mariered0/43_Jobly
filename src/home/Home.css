.Home-content {
    padding: 100px;
    text-align: center;
    height: 100vh;
}

.Home-btn {
    margin: 10px;
}

.Home-attribute {
    position: absolute;
    bottom: 0;
    left: 0;
    right: 0;
    margin-bottom: 10px;
}

