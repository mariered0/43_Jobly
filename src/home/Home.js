import React, { useContext } from "react";
import { Link } from "react-router-dom";
import { Button } from "reactstrap";
import userContext from "../user/UserContext";
import "./Home.css";

function Home() {
  const { currentUser } = useContext(userContext);

  return (
    <div className="container">
      <div className="Home-content">
        <h1 className="Home-title">Jobly</h1>
        <p className="Home-text">All the jobs in one, convenient place.</p>
        {currentUser ? (
          <h1 className="Home-title">Welcome Back, {currentUser.firstName}!</h1>
        ) : (
          <>
            <Link to="/signup">
              <Button className="Home-btn" color="primary">
                Sign Up
              </Button>
            </Link>

            <Link to="/login">
              <Button className="Home-btn" color="primary">
                Log In
              </Button>
            </Link>
          </>
        )}

        <div className="Home-attribute">
          Photo by{" "}
          <a href="https://unsplash.com/@pramodtiwari?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText">
            Pramod Tiwari
          </a>{" "}
          on{" "}
          <a href="https://unsplash.com/t/textures-patterns?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText">
            Unsplash
          </a>
        </div>
      </div>
    </div>
  );
}

export default Home;
