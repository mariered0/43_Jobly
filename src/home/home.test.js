import React from "react";
import { render } from "@testing-library/react";
import Home from "./Home";

// //importing the provider here for react context.
import UserContext from "../user/UserContext";

const currentUser = jest.fn();

it("renders without crashing", () => {
  render(
    <UserContext.Provider value={{ currentUser }}>
      <Home />
    </UserContext.Provider>
  );
});

it("matches snapshot", () => {
  const { asFragment } = render(
    <UserContext.Provider value={{ currentUser }}>
      <Home />
    </UserContext.Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("has the message", () => {
  const msg = "All the jobs in one, convenient place.";
  const { getByText } = render(
  <UserContext.Provider value={{ currentUser }}>
    <Home />
  </UserContext.Provider>
  );
  expect(getByText(msg)).toBeInTheDocument();
});
