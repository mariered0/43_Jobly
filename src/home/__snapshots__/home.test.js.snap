// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="Home-content"
    >
      <h1
        class="Home-title"
      >
        Jobly
      </h1>
      <p
        class="Home-text"
      >
        All the jobs in one, convenient place.
      </p>
      <h1
        class="Home-title"
      >
        Welcome Back, !
      </h1>
      <div
        class="Home-attribute"
      >
        Photo by 
        <a
          href="https://unsplash.com/@pramodtiwari?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText"
        >
          Pramod Tiwari
        </a>
         on 
        <a
          href="https://unsplash.com/t/textures-patterns?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText"
        >
          Unsplash
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
